workflows/ampliseq.nf

```diff
@@ -222,13 +215,46 @@ workflow AMPLISEQ {
     ch_versions = Channel.empty()
 
     //
-    // Create a channel for input read files
-    //
-    PARSE_INPUT ( params.input, is_fasta_input, single_end, params.multiple_sequencing_runs, params.extension )
-    ch_reads = PARSE_INPUT.out.reads
-    // TODO: OPTIONAL, you can use nf-validation plugin to create an input channel from the samplesheet with Channel.fromSamplesheet("input")
-    // See the documentation https://nextflow-io.github.io/nf-validation/samplesheets/fromSamplesheet/
-    // ! There is currently no tooling to help you write a sample sheet schema
+    // Create input channels
+    //
+    ch_input_fasta = Channel.empty()
+    ch_input_reads = Channel.empty()
+    if ( params.input ) {
+        // See the documentation https://nextflow-io.github.io/nf-validation/samplesheets/fromSamplesheet/
+        ch_input_reads = Channel.fromSamplesheet("input")
+            .map{ meta, readfw, readrv ->
+                meta.single_end = single_end.toBoolean()
+                def reads = single_end ? readfw : [readfw,readrv]
```

Comment:
Hi @erikrikarddaniel ,

all seems fine now, but I think that line here will cause a pipeline error when:
1. samplesheet contains only forward reads
2. `--single_end` isnt specified

before the PR here 2. wasnt needed. 
I think changing to
```suggestion
                def reads = single_end || readrv == null ? readfw : [readfw,readrv]
```
will make sample sheet input with only forward reads and without `--single_end` pass (need to test). Would that be desireable? That could lead in mixed entries though, such as some samples being treated as single ended and some as paired-end. That would crush later the pipeline I guess.

Reply:
Haven't we required `--single_end` before and not relied on what the sample sheet looked like? (I thought so, but you seem to claim otherwise, and I suppose you're right.) I think it's fine to continue with that behaviour (or introduce it here, if required) if that simplifies handling of the input sheet.

In principle, it would be nice to be able to mix, but in most cases that would mean one wants to specify sample specific primer sequences too. That's maybe overkill?

Reply:
Now I see edit2: Fine for me.

Reply:
Thanks for your feedback, I'll test how the pipeline behaves with single-end samplesheet without specifying --single_end and will add an error message if needed. Otherwise that PR is god to go (after review) I think.

> In principle, it would be nice to be able to mix, but in most cases that would mean one wants to specify sample specific primer sequences too. That's maybe overkill?

Yes theoretically that would be not bad. But that would require some refactoring, i.e. samplesheet with optional primer sequences, primers into meta map, and some trouble such as using primer sequences for taxonomic database that would need to be matched. Probably not very problematic, but for my projects not worth it. Abundance tables also can be always merged on ASV md5sums of multiple ampliseq runs or such. I think it will complicate the pipeline considerably more, but I that would need to be evaluated if someone wants to implement that ;)

Reply:
> Haven't we required --single_end before and not relied on what the sample sheet looked like? 

Indeed, I checked, that was the case.

Reply:
I added a line below this one here making sure that when there isnt any reverse read there must be single_end=true.